File: backend/app/Models/Bot.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Bot extends Model
{
    public const DEFAULT_MODEL = 'gpt-4o-mini';

    protected $fillable = ['workspace_id', 'name', 'slug', 'description', 'system_prompt', 'model', 'status',];

    public function workspace()
    {
        return $this->belongsTo(Workspace::class);
    }
    public function sources()
    {
        return $this->hasMany(BotSource::class);
    }
}

File: fastapi/app/models/bot.py
from pydantic import BaseModel, ConfigDict

from app.models.bot_source import BotSource
from app.models.workspace import Workspace


class Bot(BaseModel):
    """
    Represents a bot returned by Laravel.
    """

    model_config = ConfigDict(
        extra="forbid",
        frozen=True,
    )

    id: int

    workspace_id: int

    name: str

    description: str | None = None

    system_prompt: str | None = None

    status: str

    workspace: Workspace

    sources: list[BotSource]